server/proxy/install.go

```diff
@@ -157,6 +158,38 @@ func (p *Proxy) ensureOAuthApp(r *incoming.Request, conf config.Config, app *app
 	return nil
 }
 
+func (p *Proxy) createAndValidateBot(r *incoming.Request, bot *model.Bot) error {
+	mm := p.conf.MattermostAPI()
+
+	err := mm.Bot.Create(bot)
+	if err != nil {
+		return err
+	}
+
+	r.Log.Debugw("App install flow: created Bot Account ",
+		"username", bot.Username, "bot_user_id", bot.UserId)
+
+	const tryCount = 5
+	delay := 50 * time.Millisecond
+	i := 0
+	for {
```

Comment:
Why is this retry function needed? `mm.Bot.Create` is synchronous. In `EnsureBot` the same pattern is used: Create a bot and then update the icon. Is that also problematic?